core/dataset.cpp

```diff
@@ -703,6 +708,28 @@ bool DatasetConstProxy::operator!=(const DatasetConstProxy &other) const {
   return !dataset_equals(*this, other);
 }
 
+std::unordered_map<Dim, scipp::index> DatasetConstProxy::dimensions(
+    const std::vector<std::pair<Slice, scipp::index>> &currentSlices) const {
+
+  auto base_dims = m_dataset->dimensions();
+  // Note current slices are ordered, but NOT unique
+  for (const auto & [ slice, extents ] : currentSlices) {
+    (void)extents;
+    auto it = base_dims.find(slice.dim);
+    if (it != base_dims.end()) {
+      if (slice.end == -1) { // For non-range. Erase dimension
+        base_dims.erase(it);
+      } else {
+        it->second =
+            slice.end -
+            slice.begin; // Take extent from slice. This is the effect that
+                         // the successful slice range will have
+      }
+    }
+  }
+  return base_dims;
```

Comment:
It feels like this must be nearly duplicating code somewhere else (in the slicing logic?), but unfortunately I do not have a better suggestion right now.

Reply:
I think we should probably make another issue to look at this. Some aspect of the logic here are similar to that in the `ConstProxy` constructor.